go/chat/convsource.go

```diff
@@ -485,28 +495,76 @@ func (m ByMsgID) Len() int           { return len(m) }
 func (m ByMsgID) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
 func (m ByMsgID) Less(i, j int) bool { return m[i].GetMessageID() > m[j].GetMessageID() }
 
-func (s *HybridConversationSource) GetMessages(ctx context.Context, convID chat1.ConversationID,
-	uid gregor1.UID, msgIDs []chat1.MessageID, finalizeInfo *chat1.ConversationFinalizeInfo) ([]chat1.MessageUnboxed, error) {
-
-	rmsgsTab := make(map[chat1.MessageID]chat1.MessageUnboxed)
-
-	msgs, err := s.storage.FetchMessages(ctx, convID, uid, msgIDs)
+// getMessagesShared gets messages by ID.
+// The returned messages correspond to `getMsgIDs`.
+// All the new messages in `preFetched` are also saved.
+func (s *HybridConversationSource) getMessagesShared(
```

Comment:
Idk, I feel like they're supposed to function exactly the same way and they keep accumulating things like
- handle offline
- sort the messages
- remember to identify the tlf at the end